### DISCUSSION TITLE
mounting /home instead of /home/user

### DISCUSSION THREAD

**Discussion by Lichtmacher** [ltsp/ltsp#412] [Q&A]
Hello,
is it possible to mount /home instead of /home/user.

I would like that the users are enabled to see their respective homes and work with files as if they were logged in to the server directly.

Setup:
chrootless
Xubuntu 20.04
ltsp 21.01-1~ubuntu20.04.1

**Comment by alkisg** [comment]
With NFSv3 home, yes. To enable NFS, read: https://ltsp.org/man/ltsp-nfs/#examples
Note that NFSv3 LAN traffic is unencrypted, so SSHFS is more secure, that's why it's the default.

With SSHFS home, no. The LTSP client image doesn't have any secrets that can be used for secure SSH connections, except for the user password at login; and the user@server SSHFS connection doesn't have read permissions for all of /home.
It would be possible for the clients to generate static SSH keys based on their hardware, and to give them root access to /home/username on the server, but even so, the new file permissions would be all wrong.

With NFSv4 home, yes. And it would also be secure, as NFSv4 supports authentication and encryption. But there's no how-to for setting up NFSv4 and kerberos for LTSP yet, you'd need to do a lot of research and reading to set it up.

**Reply by Lichtmacher** [reply]
I have absolutely no experience with kerberos or LDAP. And no time. So NFSv4 is not an option for me at the moment.

I haven't completely understood yet how the sshfs is working at the moment.
Is the following right? At login (client) in lightdm the user and the password are send somehow to

/usr/share/ltsp/client/login/pamltsp

In line 196
            sshfs -o "$sshfs_params" "$@" "$pw_name@$SSH_SERVER:" "$pw_dir" 2>&1)
the /home/username of the server is mounted at /home/username in the client.

Where does the permission problems come from?

I logged into the client and did that:
mkdir /home/username/mnt_SERVER
sshfs username@SERVER:/home /home/username/mnt_SERVER

Every thing seems to be right. The files have owner:groups permissions as I expect.

What can go wrong?


Second question: If the user have an private key in their SERVER:/home/username/.ssh directory is it safe?
